osquery/tables/events/windows/windows_events.cpp

```diff
@@ -6,138 +6,246 @@
-    parseTree(node.second, results);
+Status WindowsEventSubscriber::processEventObject(
+    Event& windows_event, const boost::property_tree::ptree& event_object) {
+  windows_event = {};
+
+  Event output;
+  output.osquery_time = std::time(nullptr);
+
+  output.datetime =
+      event_object.get("Event.System.TimeCreated.<xmlattr>.SystemTime", "");
+
+  if (output.datetime.empty()) {
+    return Status::failure(
+        "Invalid Windows event object: the TimeCreated::SystemTime attribute "
+        "is missing or not valid");
+  }
+
+  output.source = event_object.get("Event.System.Channel", "");
+  if (output.source.empty()) {
+    return Status::failure(
+        "Invalid Windows event object: the Event.System.Channel tag is missing "
+        "or not valid");
   }
-  for (const auto& val : results) {
-    /// Reconstruct the event format as much as possible
-    jsonOut.put(eventDataType + "." + val.first, val.second);
+
+  output.provider_name =
+      event_object.get("Event.System.Provider.<xmlattr>.Name", "");
+
+  if (output.provider_name.empty()) {
+    return Status::failure(
+        "Invalid Windows event object: the Provider::Name attribute is missing "
+        "or not valid");
   }
 
-  std::stringstream ss;
-  boost::property_tree::write_json(ss, jsonOut, false);
+  // This field may be missing
+  output.provider_guid =
+      event_object.get("Event.System.Provider.<xmlattr>.Guid", "");
+
+  output.event_id = event_object.get("Event.System.EventID", -1);
+  if (output.event_id == -1) {
+    return Status::failure(
+        "Invalid Windows event object: the System.EventID tag is missing or "
+        "not valid");
+  }
+
+  output.task_id = event_object.get("Event.System.Task", -1);
+  if (output.task_id == -1) {
+    return Status::failure(
+        "Invalid Windows event object: the System.Task tag is missing or not "
+        "valid");
+  }
 
-  auto s = ss.str();
-  if (s.at(s.size() - 1) == '\n') {
-    s.erase(s.end() - 1);
+  output.level = event_object.get("Event.System.Level", -1);
+  if (output.level == -1) {
+    return Status::failure(
+        "Invalid Windows event object: the System.Level tag is missing or not "
+        "valid");
   }
-  r["data"] = s;
 
-  add(r);
+  auto string_keywords = event_object.get("Event.System.Keywords", "");
+  if (string_keywords.empty()) {
+    output.keywords = 0;
+
+  } else {
+    char* expected_null_term{nullptr};
+    output.keywords =
+        std::strtoull(string_keywords.c_str(), &expected_null_term, 0);
+
+    if (expected_null_term == nullptr || *expected_null_term != 0) {
+      return Status::failure(
+          "Invalid Windows event object: the System.Keywords tag is not a "
+          "valid integer value: " +
+          string_keywords);
+    }
+  }
+
+  auto event_data_node_opt = event_object.get_child_optional("Event.EventData");
+
+  if (event_data_node_opt) {
+    const auto& event_data_node = event_data_node_opt.value();
+
+    try {
```

Comment:
Here as well with the big `try`/`except`s. Any chance we can be more precise with where we're catching exceptions? I'd much rather see an exception or crash happen later than to have it bundled up with a json parsing message, when that's unrelated.